packages/language-server/index.js

```diff
@@ -54,9 +54,15 @@ connection.onInitialize(async (parameters) => {
     createTypeScriptProject(
       typescript,
       diagnosticMessages,
-      (serviceEnv, {configFileName}) => getLanguagePlugins(configFileName)
+      async ({configFileName}) => ({
+        languagePlugins: await getLanguagePlugins(configFileName)
+      })
     ),
-    getLanguageServicePlugins()
+    getLanguageServicePlugins(),
+    {
+      pullModelDiagnostics:
+        parameters.initializationOptions?.pullModelDiagnostics
```

Comment:
What does this option mean? To use pull model diagnostics instead of the default push model diagnostics? If we add a new initialization option, it should also be added to the language server readme.

Reply:
Enabling `pullModelDiagnostics` is to solve the problem of `serverHandle.sendDocumentDiagnosticRequest` not responding.

By default, diagnostics are pushed by the language server. At this time, `initializeResult.capabilities.diagnosticProvider` will be set to undefined.

https://github.com/volarjs/volar.js/blob/ea2e22ca17116b4274d661a014cff040c1d206cb/packages/language-server/lib/server.ts#L197

In 2.4, when `initializeResult.capabilities.diagnosticProvider` is `undefined`, the language server will not pass through the `server.connection.languages.diagnostics.on` event, causing `serverHandle.sendDocumentDiagnosticRequest` to not respond.

https://github.com/volarjs/volar.js/blob/ea2e22ca17116b4274d661a014cff040c1d206cb/packages/language-server/lib/register/registerLanguageFeatures.ts#L357

Since this is dedicated to internal testing, I don't think it needs to be added to the language server readme

Reply:
After more consideration, I believe that `server.connection.languages.diagnostics.on` should not be registered based on whether `initializeResult.capabilities.diagnosticProvider` is true, but rather on whether any LanguageServicePlugin that supports diagnostics is provided.

In the refactoring of https://github.com/volarjs/volar.js/pull/230, `pullModelDiagnostics` has been removed, so this change no longer exists!